### PR TITLE
Remove `Component` suffix from `AutoComplete` + directory organization

### PR DESCRIPTION
**Pull request by manuelpuyol**
This PR removes the `Component` suffix from `AutoComplete` and `AutoCompleteItem`.
It also changes the file structure for this component:
```
auto_complete.rb
auto_complete/
├── auto_complete.html.erb
├── auto_complete.d.ts
├── auto_complete.js
├── auto_complete.ts
└── item.rb
```
This directory structure makes the project more organized and easier to navigate imo.

I'd really like to move the `.rb` file inside the folder, but I'm not sure how to do it without adding a new suffix since zeitwerk will expect the class to be `Primer::AutoComplete::FileName`. In the future we could also move `preview` and `stories` to be in the same folder!

### CODE REVIEW

**Review by BlakeWilliams — APPROVED** [review]
One comment about the `short_name` logic, but LGTM!

File: Rakefile [hunk 1]
```diff
@@ -88,7 +89,7 @@ namespace :docs do
   end
 
   def link_to_component(component)
-    short_name = component.name.demodulize.gsub("Component", "")
+    short_name = component.name.gsub(/Primer|::|Component/, "")
```
[the remark this answers is not in the document]
Comment (manuelpuyol) [reply Rakefile:92]: problem with `demodulize` is that `Primer::AutoComplete::Item` would become `Item` instead of `AutoCompleteItem` :/

**Comment by manuelpuyol** [comment]
thanks for the review @BlakeWilliams 
you have any ideas how I could put the `.rb` file inside the directory without having the class be `Primer::AutoComplete::AutoComplete`? 😬

**Comment by BlakeWilliams** [comment]
We chatted a bit in Slack, but I think https://github.com/fxn/zeitwerk#collapsing-directories might help here?

That might be more effort to maintain though.

File: Rakefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (BlakeWilliams) [reply Rakefile:92]: Oh I see! Because `Item` is now namespaced too. That makes sense to me. 👍

**Pull request merged**